### PR TITLE
Improves coverage for the aws_cryptosdk_enc_ctx_init proof 

### PR DESCRIPTION
**Pull request by tegansb**
Description of changes:
Addresses the unexpected missing function in the aws_cryptosdk_enc_ctx_init proof.
Includes a README for the aws_cryptosdk_enc_ctx_init proof specifying expected coverage behavior.
Adds missing dependencies tot he Makefile. 

Coverage for aws_cryptosdk_enc_ctx_init improves from .89 to .91. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

@danielsn @feliperodri 

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

File: .cbmc-batch/jobs/aws_cryptosdk_enc_ctx_init/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Memory safety proof for aws_cryptosdk_enc_ctx_init
+
+This proof harness attains 91% code coverage.  The following comments explain
+why the uncovered lines of code are unreachable code.
+
+Some functions contain unreachable blocks of code:
```
Comment (danielsn) [inline .cbmc-batch/jobs/aws_cryptosdk_enc_ctx_init/README.md:6]: Are these intrinsic to the function, or limitations of the harness?

Comment (tegansb) [reply .cbmc-batch/jobs/aws_cryptosdk_enc_ctx_init/README.md:6]: These are intrinsic to the function the way it is currently written. I have added more details in the README about why code is unreachable.

Comment (tawdry-audrey) [reply .cbmc-batch/jobs/aws_cryptosdk_enc_ctx_init/README.md:6]: Sounds good, and the documentation will help with future efforts!

File: .cbmc-batch/jobs/aws_cryptosdk_enc_ctx_init/aws_cryptosdk_enc_ctx_init_harness.c [hunk 2]
```diff
@@ -16,8 +16,7 @@
 #include <aws/cryptosdk/enc_ctx.h>
 #include <proof_helpers/proof_allocators.h>
 
-// This is a memory safety proof for aws_cryptosdk_multi_keyring() defined in
```
Comment (danielsn) [inline .cbmc-batch/jobs/aws_cryptosdk_enc_ctx_init/aws_cryptosdk_enc_ctx_init_harness.c:19]: good catch

**Review by danielsn — APPROVED** [review]

**Pull request merged**